neural_compressor/adaptor/pytorch.py

```diff
@@ -4434,6 +4457,192 @@ def calculate_op_sensitivity(self, model, dataloader, tune_cfg, output_op_names,
+        """
+        assert isinstance(model._model, torch.nn.Module), \
+               "The model passed in is not the instance of torch.nn.Module"
+        if self.performance_only:
+            q_model = model
+        else:
+            try:
+                q_model = copy.deepcopy(model)
+            except Exception as e:  # pragma: no cover
+                logger.warning("Fail to deep copy the model due to {}, inplace is used now.".format(
+                    repr(e)))
+                q_model = model
+
+        # For tensorboard display
+        self.tune_cfg = tune_cfg
+        self.tune_cfg["approach"] = self.approach
+        self.tune_cfg["framework"] = "pytorch"
+        assert self.approach=='post_training_weight_only', "Please make sure the approach is weight_only"
+
+        q_model._model = self.RTN_quantize(q_model._model, tune_cfg)
+        q_model._model = self.GPTQ_quantize(q_model._model, tune_cfg, dataloader)
+        q_model._model = self.AWQ_quantize(q_model._model, tune_cfg, dataloader)
+
+        q_model.q_config = copy.deepcopy(self.tune_cfg)
+        q_model.is_quantized = True
+        self._dump_model_op_stats(q_model._model, q_model.q_config)
+        return q_model
+
+    def RTN_quantize(self, model, tune_cfg):
+        from .torch_utils.weight_only import quant_weight
+        from .torch_utils.util import fetch_module
+        for key, config in tune_cfg['op'].items():
+            op_name, op_type = key
+            if config['weight']['dtype'] == 'fp32':
+                continue
+            else:
+                num_bits = config['weight']['bit']
+                group_size = config['weight']['group_size']
+                scheme = config['weight']['scheme']
+                algorithm = config['weight']['algorithm']
+                if algorithm != 'RTN':
+                    continue
+                m = fetch_module(model, op_name)
+                weight = m.weight
+                if op_type == "Conv2d":
+                    orig_shape = weight.shape
+                    weight = weight.permute(1, 0, 2, 3)
+                    weight = weight.reshape(weight.shape[0], -1)
+                q_weight = quant_weight(weight, num_bits, group_size, scheme)
+                if op_type == "Conv2d":
+                    q_weight = q_weight.reshape(orig_shape[1], orig_shape[0], orig_shape[2], orig_shape[3])
+                    q_weight = q_weight.permute(1, 0, 2, 3)
+                m.weight.data.copy_(q_weight)
+        return model
+
+    def GPTQ_quantize(self, model, tune_cfg, dataloader):
+        if 'gptq_args' in self.recipes:
+            percdamp = self.recipes['gptq_args'].get('percdamp', 0.01)
+        # GPTQ(model, dataloader, w_bit, group_size, percdamp=0.01)
+        # TODO: implementation
+        return model
+
+    def AWQ_quantize(self, model, tune_cfg, dataloader):
+        # set default value if has args in recipes, else we use function 
+        if 'gptq_args' in self.recipes:
```

Comment:
This is a wrong args. To avoid double pre-ci, I will fix it after this PR is merged.